casanovo/denovo/model.py

```diff
@@ -444,6 +461,37 @@ def _terminate_finished_beams(
                             abs(d) < self.precursor_mass_tol
                             for d in delta_mass_ppm
                         )
+
+                        # Don't terminate if m/z difference is by NH3 loss
+                        if (
+                            exceeds_precursor_mz_tol
+                            and not is_within_precursor_mz_tol
+                            and is_nh3_loss_in_vocab
+                        ):
+                            alt_peptide_seq = peptide_seq + ["-17.027"]
+                            pred_mz = self.peptide_mass_calculator.mass(
+                                seq=alt_peptide_seq, charge=precursor_charge
+                            )
+                            delta_mass_ppm = [
+                                _calc_mass_error(
+                                    pred_mz,
+                                    precursor_mz,
+                                    precursor_charge,
+                                    isotope,
+                                )
+                                for isotope in range(
+                                    self.isotope_error_range[0],
+                                    self.isotope_error_range[1] + 1,
+                                )
+                            ]
+                            is_precursor_mz_tol_w_loss = any(
+                                abs(d) < self.precursor_mass_tol
+                                for d in delta_mass_ppm
+                            )
+                            exceeds_precursor_mz_tol = (
```

Comment:
**nitpick:** Combine the previous statement with this one, i.e. `not any(...)`. Just doing `not` of another boolean is a bit silly.